src/FlanCommand.ts

```diff
@@ -57,39 +58,86 @@ export default abstract class FlanCommand extends Command {
-    let configJSON;
+    // get envirment variable configurations
+    let envConfig: {
+      baseDir?: string;
+      saveDir?: string;
+      repoDir?: string;
+      repository?: string;
+    } = {};
+    let envDbConfig: { host?: string; db?: string; user?: string } = {};
+
+    if (process.env.FLAN_DB_HOST) {
+      envDbConfig = { ...envDbConfig, host: process.env.FLAN_DB_HOST };
+    }
+    if (process.env.FLAN_DB_NAME) {
+      envDbConfig = { ...envDbConfig, db: process.env.FLAN_DB_NAME };
+    }
+    if (process.env.FLAN_DB_USER) {
+      envDbConfig = { ...envDbConfig, user: process.env.FLAN_DB_USER };
+    }
+    if (process.env.FLAN_BASE_DIR) {
+      envConfig = { ...envConfig, baseDir: process.env.FLAN_BASE_DIR };
+    }
+    if (process.env.FLAN_SAVE_DIR) {
+      envConfig = { ...envConfig, saveDir: process.env.FLAN_SAVE_DIR };
+    }
+    if (process.env.FLAN_REPO_DIR) {
+      envConfig = { ...envConfig, repoDir: process.env.FLAN_REPO_DIR };
+    }
+    if (process.env.FLAN_REPOSITORY) {
+      envConfig = { ...envConfig, repository: process.env.FLAN_REPOSITORY };
+    }
 
+    // load config file
+    let configJSON: {
+      database?: { host?: string; db?: string; user?: string };
+      baseDir?: string;
+      saveDir?: string;
+      repoDir?: string;
+      repository?: string;
+    } = {};
     try {
       configJSON = await fs.readJson(configPath);
     } catch (error) {
-      this.error("Unable to load configuration");
+      if (!error.code || error.code !== "ENOENT") {
+        this.error("A config file exists but it could not be loaded");
+      }
     }
 
     this.localConfig = {
       ...this.localConfig,
       ...configJSON,
-      baseDir: path.resolve(configJSON.baseDir || this.localConfig.baseDir),
+      ...envConfig,
+      baseDir: path.resolve(
+        envConfig.baseDir || configJSON.baseDir || this.localConfig.baseDir
+      ),
       // set full paths
       repoDir: path.resolve(
-        configJSON.baseDir || this.localConfig.baseDir,
+        envConfig.baseDir || configJSON.baseDir || this.localConfig.baseDir,
         "./repo"
       ),
       saveDir: path.resolve(
-        configJSON.baseDir || this.localConfig.baseDir,
+        envConfig.baseDir || configJSON.baseDir || this.localConfig.baseDir,
         "./local"
       ),
       database: {
         ...this.localConfig.database,
         ...configJSON.database,
+        ...envDbConfig,
       },
     };
 
-    if (configJSON.repoDir) {
-      this.localConfig.repoDir = path.resolve(configJSON.repoDir);
+    if (envConfig.repoDir || configJSON.repoDir) {
+      this.localConfig.repoDir = path.resolve(
+        envConfig.repoDir || configJSON.repoDir || ""
+      );
     }
 
-    if (configJSON.saveDir) {
-      this.localConfig.saveDir = path.resolve(configJSON.saveDir);
+    if (envConfig.saveDir || configJSON.saveDir) {
+      this.localConfig.saveDir = path.resolve(
+        envConfig.saveDir || configJSON.saveDir || ""
```

Comment:
yeah both of these should have a value according to the conditional, not sure if `|| ""` is needed?

Reply:
So I tried removing it but it seems that TypeScript is not smart enough to figure out that the statement in the path.resolve matches the if statement. When I remove the `|| ""` VS code complains about the parameter possibly being undefined. I agree that the `"" ` should not be needed but I also don't see a cleaner way to get rid of it without tripping up TypeScript.

Reply:
haha that's definitely annoying, if you throw a comment above each one then we should good to go